7. Dynamic Programming/spring_throw/SpringThrowArnab.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * SpringThrowArnab.java
+ * given an array of spring and blocks with their ranges of throws.Calculate the minimum number of spring throws required to reach the end of the blocks
+ * Description:-
+ * Just check for every spring whats the maximum distance it can send the person,in this way update the current position,and keep 
+ * on calculating max.When j reaches that current position,one cycle of spring throw is completed,hence update the count by 1
+ * And then make current to the next maximum uptill then.That's how it will continue uptill n-1.As there it will always be ending 
+ * at the last block by any means
+ * Time Complexity-O(n) Extra Space Complexity-O(1)
+ * @author [codebook-2000](https://github.com/codebook-2000)
+ */
+ 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+public class SpringThrowArnab {
+    static int solve(int[] arr, int n) {
+        //ct stores the count of no of throws
+        //max stores the maximum distance upto which every spring can send the person
+        //current is the point uptill where person can go by the string after which it needs to get updated
+
+        int j = 0, ct = 0, max = 0, current = 0;
+        while (j < n - 1) {  //calculating till n-2 as after it it will directly be at last index
+            max = Math.max(max, arr[j] + j);//storing the maximum of all distances it can travel
```

Comment:
In your solution, I believe you are not checking if the block as reached the nth block or not, but rather if it is reaching nth block and beyond. Can you check the problem statement as it requires you to reach the nth block and not beyond.